part1-장현태-sprint4/src/services/product.js

```diff
@@ -0,0 +1,46 @@
+import productRepository from "../repositories/productRepository.js";
+import likeRepository from "../repositories/likeRepository.js";
+
+export async function fetchProducts(query) {
+  return await productRepository.getAll(query);
+}
+
+export async function getProductById(id) {
+  return await productRepository.getById(id);
+}
+
+export async function createNewProduct(userId, productData) {
+  return await productRepository.save(userId, productData);
+}
+
+export async function updateProductById(id, product) {
+  const NewproductData = {
+    name: product.name,
+    description: product.description,
+    price: product.price,
+    tags: product.tags,
+  };
+  return await productRepository.update(id, NewproductData);
+}
+
+export async function removeProductById(id) {
+  return await productRepository.deleteById(id);
+}
+
+export async function addProductLike(userId, productId) {
+  await likeRepository.saveProductLike(userId, productId);
+  return;
+}
+
+export async function cancelProductLike(userId, productId) {
+  await likeRepository.deleteProductLike(userId, productId);
+  return;
+}
+
+export async function checkIsLikedByUser(userId, productId) {
+  const LikedProducts = await productRepository.getAllLikedProduct(userId);
+  for (const product of LikedProducts) {
+    if (product.id === productId) return true;
+    return false;
```

Comment:
이렇게 수정 할 수도 있을 것 같아요
```suggestion
    return product.id === productId
```